Sources/SparkConnect/ErrorUtils.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+import Foundation
+
+/// Utility functions like `org.apache.spark.util.SparkErrorUtils`.
+public enum ErrorUtils {
+  public static func tryWithSafeFinally<T>(
+    _ block: () async throws -> T, _ finallyBlock: () async throws -> Void
+  ) async rethrows -> T {
+    let result: T
+    do {
+      result = try await block()
+      try await finallyBlock()
+    } catch {
+      try await finallyBlock()
```

Comment:
```suggestion
      try? await finallyBlock()
```

Reply:
Perhaps this is closer to the exception handling approach of `tryWithSafeFinally` in the Scala version?

Reply:
I agree with you. Thank you.